#187 Fix logging configuration for JAR product

ndoschek opened the pull request:
- Update log configuration
  - If file path is given, reconfigure logging based on this configuration
  - Default case configures console logging to level INFO, no file logging
- Add ExampleServerLauncher.launch configuration that uses the example log configuration (console and file logging at level info)
- Remove log configuration files from classpath to avoid unwanted configuration aggregation
- Fix performance warning that causes log42j logging issue if executed in bundled jar
- Switch to import instead of require plugin bundles for log4j dependencies
- Update lib dependencies
  - Do not exclude source libs to improve debugging experiences in p2 use case
- Update README
  - Update logging section
  - Format via markdownlint and remove markdownlint warnings
  - Update import project section and remove outdated information
- targetplatform project
  - Include targetplatform as module of releng parent project to be able to import all projects as maven projects without exception
  - Remove unused targetplatform profile

Part of #187

Please also have a look at https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/pull/110

cdamus commented on bundles/org.eclipse.emfcloud.modelserver.common/META-INF/MANIFEST.MF:
```diff
@@ -20,4 +19,5 @@ Export-Package: org.eclipse.emfcloud.modelserver.common,
 Bundle-Vendor: EclipseSource
-Import-Package: com.google.inject.multibindings;version="1.3.0"
+Import-Package: com.google.inject.multibindings;version="1.3.0",
+ org.apache.logging.log4j;version="2.17.1"
```
Should this not be a version range?

cdamus reviewed (changes requested):
Wow, that is a thorough clean-up for developer usability. Thanks!

I do have a few quibbles in comments in-line. Also, there's a problem in the formatting in the new README:

<img width="920" alt="CleanShot 2022-07-07 at 18 25 45@2x" src="https://user-images.githubusercontent.com/1615558/177886199-75a20da8-4573-4e3d-8c41-8d91fe1123c1.png">

This section lost the fence around the code block and all of the indentation within it.

ndoschek commented on bundles/org.eclipse.emfcloud.modelserver.common/META-INF/MANIFEST.MF:
```diff
@@ -20,4 +19,5 @@ Export-Package: org.eclipse.emfcloud.modelserver.common,
 Bundle-Vendor: EclipseSource
-Import-Package: com.google.inject.multibindings;version="1.3.0"
+Import-Package: com.google.inject.multibindings;version="1.3.0",
+ org.apache.logging.log4j;version="2.17.1"
```
That's a good point, thanks for catching that!

ndoschek commented:
Thanks a lot for you review @cdamus, I addressed your comments with a separate commit, would be great if you could have another quick look, thanks again!

cdamus commented on bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/launch/ModelServerLauncher.java:
```diff
@@ -125,10 +124,10 @@ protected void configure() {
-   public static void configureLogger(final Optional<String> configurationFilePath) {
-      if (configurationFilePath.isPresent()) {
+   protected static void configureLogger(final String configurationFilePath) {
+      if (configurationFilePath != null && !configurationFilePath.isEmpty()) {
```
Good catch on the present-but-empty string! 😀
